Lib/http/server.py

```diff
@@ -769,9 +775,32 @@ def send_head(self):
                             f.close()
                             return None
 
-            self.send_response(HTTPStatus.OK)
+            if self._range:
+                start, end = self._range
+                if start is None:
+                    # `end` here means suffix length
+                    start = max(0, fs.st_size - end)
```

Comment:
Add a comment saying that `parse_range()` collapses `(None, None)` to `None` and thus `end` cannot be None (you can also add an assertion).